### PR TITLE
fix(referentialQuestions): fix the referential questions changing not…

### PR DESCRIPTION
**Pull request by Theo-lbg**
… working

The update working well but in the UI, we cannot see it.

In this PR : 

- We have test if the referential was been updated.
- We have test if we can updated two referential one after the other

Tests created.

### CODE REVIEW

File: lib/presentation/pages/qvst/widgets/response_reference_widget.dart [hunk 1]
```diff
@@ -22,32 +25,41 @@ class ResponseReferenceWidget extends ConsumerStatefulWidget {
...
+  QvstAnswerRepoEntity? _initialRepo;
+  bool _isSaved = true;
+
+  void _initializeRepo(List<QvstAnswerRepoEntity> data) {
+    if (_initialRepo == null) {
+      _initialRepo =
+          data.firstWhere((element) => element.id == widget.referenceId);
+      _selectedRepo = _initialRepo;
+    }
+  }
```
Comment (PiotrFLEURY) [inline lib/presentation/pages/qvst/widgets/response_reference_widget.dart:37]: prefer use provider arguments instead

see [Passing arguments to provider](https://riverpod.dev/fr/docs/essentials/passing_args)

Should be

```dart
final selectedRepo = ref.watch(qvstAnswerRepoListProvider(widget.referenceId));
``` 

init data when you receive other data is not recommended and can cause poor performances on UI

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]
Please consider previous comment

**Review by PiotrFLEURY — APPROVED** [review]
LGTM

**Review by NyraSama — APPROVED** [review]
LGTM

**Pull request merged**